### PR TITLE
ref(notifications): Revert-Revert Simplify MessageBuilder hierarchy

### PR DESCRIPTION
**Pull request by mgaeta**
This is a revert-revert of https://github.com/getsentry/sentry/pull/29721.

Updated getsentry tests: https://github.com/getsentry/getsentry/pull/6572.

### CODE REVIEW

**Comment by scefali** [comment]
@mgaeta where is the getsentry PR that goes with these changes?

File: src/sentry/api/helpers/group_index/update.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def update_groups(
                 .order_by("-sort")[0]
             )
             activity_type = Activity.SET_RESOLVED_IN_RELEASE
-            activity_data: MutableMapping[str, Optional[Any]] = {
+            activity_data: MutableMapping[str, Any | None] = {
```
Comment (scefali) [inline src/sentry/api/helpers/group_index/update.py:253]: @mgaeta were these changes caused by linting? Or manual changes?

Comment (mgaeta) [reply src/sentry/api/helpers/group_index/update.py:253]: This change was automatically applied by the git-commit hook `pyupgrade`.

File: src/sentry/integrations/slack/message_builder/notifications.py [hunk 2]
```diff
@@ -1,109 +1,62 @@
...
-                        "url": project_url,
-                    }
-                )
-    return buttons
+def get_message_builder(klass: str) -> type[SlackNotificationsMessageBuilder]:
+    """TODO(mgaeta): HACK to get around circular imports."""
+    return {
+        "IssueNotificationMessageBuilder": IssueNotificationMessageBuilder,
+        "SlackNotificationsMessageBuilder": SlackNotificationsMessageBuilder,
+    }[klass]
```
Comment (scefali) [inline src/sentry/integrations/slack/message_builder/notifications.py:17]: @mgaeta I should point out this change will prevent any class in `getsentry` from making its own Slack message builder. Not an issue right now but if we ever needed a custom message builder, we'll have to figure out another solution here.

File: src/sentry/integrations/slack/message_builder/notifications.py [hunk 3]
```diff
@@ -1,109 +1,62 @@
...
-
-
-def build_notification_attachment(
-    notification: BaseNotification,
-    context: Mapping[str, Any],
-    recipient: Union["Team", "User"],
-) -> SlackBody:
-    """@deprecated"""
-    return notification.build_slack_attachment(context, recipient)
+        return SlackIssuesMessageBuilder(
```
Comment (scefali) [inline src/sentry/integrations/slack/message_builder/notifications.py:54]: @mgaeta Is this going to be a temporary solution or is this a pattern you plan on sticking with? Cause I don't think having message builders call each other is a great long-term solution.

**Review by scefali — APPROVED** [review]

**Pull request merged**